oap-server/server-library/library-client/src/main/java/org/apache/skywalking/oap/server/library/client/elasticsearch/ElasticSearchClient.java

```diff
@@ -331,9 +333,25 @@ public boolean deleteTemplate(String indexName) throws IOException {
         return response.getStatusLine().getStatusCode() == HttpStatus.SC_OK;
     }
 
+    public SearchResponse search(IndexNameMaker indexNameMaker, SearchSourceBuilder searchSourceBuilder) throws IOException {
+        String[] indexNames = Arrays.stream(indexNameMaker.make()).map(this::formatIndexName).toArray(String[]::new);
+        return search(indexNames, searchSourceBuilder);
```

Comment:
Why need and call `search` again? And in both methods, you are only doing the `formatIndexName`. I think the namespace would be added twice.

Reply:
What happens if the `indexNames#size` == 0?

Reply:
1. get real index name by String[] indexNames = Arrays.stream(indexNameMaker.make()).map(this::formatIndexName).toArray(String[]::new);
2. do search search(indexNames, searchSourceBuilder) input is real indexNames. 

in TimeSeriesUtils, the default name is alias name ,so the input would not be empty

Reply:
You are calling another `search`, not `doSearch`. That is what I am asking. Please recheck the codes.

Reply:
sorry

Reply:
Do we still need another `search(array)`?